### PR TITLE
Fix clippy lints

### PR DESCRIPTION
**Pull request by carloabelli**
This fixes all but 1 clippy lint. Most of the changes are simple. Biggest change is breaking the net block update function up into smaller functions.

Perhaps it might be worth adding automatic clippy linting to CI?

### CODE REVIEW

**Comment by ammgws** [comment]
So all those `clone()`s were safe to remove in the end?
https://rust-lang.github.io/rust-clippy/master/#redundant_clone

Also, can you submit the net block code refactor as another PR?

File: src/blocks/sound.rs [hunk 1]
```diff
@@ -724,7 +724,7 @@ impl Sound {
             self.text.set_text(if self.bar {
                 format_percent_bar(volume as f32)
             } else {
-                text.to_string()
+                text
```
Comment (ammgws) [inline src/blocks/sound.rs:727]: What lint was this one? Surprised I didn't see it before as I thought I had got all the trivial ones like this already.

Comment (carloabelli) [reply src/blocks/sound.rs:727]: this was also a redundant clone

**Comment by carloabelli** [comment]
> So all those `clone()`s were safe to remove in the end?

They should be safe to remove. For most of them the function takes ownership and it can just move it into the next function.

> Also, can you submit the net block code refactor as another PR?

Done.

**Comment by ammgws** [comment]
> Perhaps it might be worth adding automatic clippy linting to CI?

Yeah, though perhaps we should start with simple/conservative lints so new contributors don't get too discouraged at first.

**Comment by carloabelli** [comment]
> Yeah, though perhaps we should start with simple/conservative lints so new contributors don't get too discouraged at first.

I see your point here. If clippy is enabled in CI, it would be good to limit lints to balance this.

I will say that in my experience the recommended fix for this lint has been correct. It can sometimes require some reasoning about how all the borrowing works to really understand why, but for a new contributor just going with the recommended fix, it should work almost every time.

**Comment by ammgws** [comment]
Send a PR for enabling CI clippy lints?

**Pull request merged**